src/ProjectService.Data/TaskRepository.cs

```diff
@@ -48,18 +49,44 @@ public TaskRepository(IDataProvider provider)
             _provider = provider;
         }
 
+        public Guid CreateTask(DbTask newTask)
+        {
+            _provider.Tasks.Add(newTask);
+            _provider.Save();
+
+            return newTask.Id;
+        }
+
+        public bool IsExist(Guid id)
+        {
+            return _provider.Tasks.FirstOrDefault(x => x.ParentId == id) != null;
+        }
+
         public bool Edit(DbTask task, JsonPatchDocument<DbTask> taskPatch)
         {
             taskPatch.ApplyTo(task);
             _provider.Save();
 
             return true;
         }
-
         public DbTask Get(Guid taskId)
         {
             return _provider.Tasks.FirstOrDefault(x => x.Id == taskId) ??
-                   throw new NotFoundException($"Task id '{taskId}' was not found.");
+                throw new NotFoundException($"Task id '{taskId}' was not found.");
+        }
+
+        public ConcurrentDictionary<Guid, int> MaxNumber()
+        {
+            var dictionary = new ConcurrentDictionary<Guid, int>();
+            var projectIds = _provider.Projects.Select(x => x.Id);
+            foreach (var Id in projectIds)
```

Comment:
Id is local variable => name it id and rename to projctId

Reply:
done